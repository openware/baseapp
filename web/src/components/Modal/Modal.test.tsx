import {shallow, ShallowWrapper} from 'enzyme';
import * as React from 'react';
import { Button } from 'react-bootstrap';
import { Modal, ModalProps } from './';

const defaults: ModalProps = {
    show: true,
    header: <div>Title</div>,
    content: <div>Some content</div>,
    footer: <Button onClick={jest.fn()} />,
};

const setup = (props: Partial<ModalProps> = {}) =>
    shallow(<Modal {...{ ...defaults, ...props }} />);

describe('Basic Modal', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = setup();
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        expect(wrapper.hasClass('cr-modal')).toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'new-class';
        const wrapper = setup({ className });//tslint:disable-line
        expect(wrapper.hasClass(className)).toBeTruthy();
    });

    it('should handle false value in show prop', () => {
        expect(wrapper.props()).not.toBeNull();
        wrapper = setup({ show: false });
        expect(wrapper.props()).toEqual({});
    });
});
