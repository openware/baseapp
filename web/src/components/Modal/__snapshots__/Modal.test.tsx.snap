// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Modal should render 1`] = `
<div
  className="cr-modal"
>
  <div
    className="cr-modal__container"
  >
    <div
      className="cr-modal__container-header"
    >
      <div>
        Title
      </div>
    </div>
    <div
      className="cr-modal__container-content"
    >
      <div>
        Some content
      </div>
    </div>
    <div
      className="cr-modal__container-footer"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[MockFunction]}
        variant="primary"
      />
    </div>
  </div>
</div>
`;
